### PR TITLE
Allow remote mapping to UWP / Immersive Windows MR apps

### PR DESCRIPTION
**Pull request by n01d**
Extend the RemoteMeshTarget so it can not only be used in the Unity Editor to receive a mesh from a Hololens but also within UWP Immersive Windows Mixed Reality applications. Also allow the saving and loading of a mesh file from an UWP app.

### CODE REVIEW

File: Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs [hunk 1]
```diff
@@ -36,10 +192,7 @@ public class RemoteMeshTarget : SpatialMappingSource
         /// </summary>
         private TcpClient networkClient;
 
-        /// <summary>
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs]: Please revert the summary removal.

File: Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/FileSurfaceObserver.cs [hunk 2]
```diff
@@ -4,7 +4,11 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-namespace MixedRealityToolkit.SpatialMapping.RemoteMapping
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace HoloToolkit.Unity.SpatialMapping
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/FileSurfaceObserver.cs]: Revert the namespace change for the Dev Branch, or retarget May or June Releases

File: Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMappingManager.cs [hunk 3]
```diff
@@ -10,7 +9,7 @@
 using UnityEngine.Windows.Speech;
 #endif
 
-namespace MixedRealityToolkit.SpatialMapping.RemoteMapping
+namespace HoloToolkit.Unity.SpatialMapping
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMappingManager.cs]: Revert the namespace change for the Dev Branch, or retarget May or June Releases

File: Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs [hunk 4]
```diff
@@ -4,13 +4,18 @@
...
 using System.IO;
+using UnityEngine;
+
 #if UNITY_EDITOR || UNITY_STANDALONE_WIN
 using System.Net;
 using System.Net.Sockets;
 #endif
-using UnityEngine;
 
-namespace MixedRealityToolkit.SpatialMapping.RemoteMapping
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/RemoteMeshTarget.cs]: Revert the namespace change for the Dev Branch, or retarget May or June Releases

**Comment by n01d** [comment]
Reverted the namespace changes and added deleted summary

**Comment by n01d** [comment]
Any update? Did I miss something??

**Comment by SimonDarksideJ** [comment]
Changes look good but agree with @StephenHodgson on those last few changes

**Comment by StephenHodgson** [comment]
@n01d any updates?

File: Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/FileSurfaceObserver.cs [hunk 5]
```diff
@@ -71,4 +75,21 @@ private void Update()
 #endif
         }
     }
+
+#if UNITY_EDITOR
+    [CustomEditor(typeof(FileSurfaceObserver))]
+    public class FileSurfaceObserverEditor : Editor
```
[the remark this answers is not in the document]
Comment (n01d) [reply Assets/MixedRealityToolkit/SpatialMapping/Scripts/RemoteMapping/FileSurfaceObserver.cs]: I added this back from a previous version of the file. This wasn't my code but I used it for testing. I will remove this again.

**Comment by n01d** [comment]
I did a suggested changes. Sorry for the late reply.

**Pull request merged**